Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -1706,7 +1706,8 @@ namespace ScriptCanvasEditor
         
         while (!isValidFileName)
         {
-            selectedFile = AzQtComponents::FileDialog::GetSaveFileName(this, tr("Save As..."), suggestedDirectoryPath.data(), filter);
+            selectedFile = QFileDialog::getSaveFileName
+                (nullptr, QObject::tr("Save As..."), suggestedDirectoryPath.data(), QObject::tr("All ScriptCanvas Files (*.scriptcanvas)"));
```

Comment:
Any reason why you can't use the AzQtComponents version? We could probably add a different wrapper if you have different needs

Reply:
+1 The AzQtComponents version was created to move some logic checks to a central location:
* Prevent user from saving to a file-name that the AP would consider invalid
* Handle missing extension when using `QFileDialog::getSaveFileName` on Linux

So we should instead fix what isn't working with `AzQtComponents::FileDialog::GetSaveFileName`. If you just pass the `filter` would it work?

Also the `this` parent should continue to be passed.